File: Leetcode/Weekly Contest 238/WC238-B.py
class Solution(object):
    def maxFrequency(self, nums, k):
        res, sum, s = 0, 0, 0
        sorted(nums)
        for i in range(len(nums)):
            sum += nums[i]
            while s < i and sum + k < nums[i] * (i - s + 1):
                sum -= nums[s]
                s += 1
            res = max(res, i - s + 1)
        return res

File: Leetcode/Weekly Contest 238/WC238-B.cpp
class Solution {
public:
	int maxFrequency(vector<int>& nums, int k) {
		sort(nums.begin(), nums.end());

		long long res = 0, sum = 0, s = 0;
		for (int i = 0; i < nums.size(); i++)
		{
			sum += nums[i];
			while (s < i && sum + k < nums[i] * (i - s + 1))
				sum -= nums[s++];
			res = max(res, i - s + 1);
		}
		return res;
	}
};